src/components/TimeCardPage/CellTypes/CommentCell.tsx

```diff
@@ -1,43 +1,219 @@
+} from "@chakra-ui/icons";
 
-import {CommentSchema} from '../../../schemas/RowSchema'; 
-import {CommentType} from '../types'; 
-import {TIMEZONE} from '../../../constants'; 
+import { CommentSchema } from "../../../schemas/RowSchema";
+import { CommentType, CellStatus, Color } from "../types";
+import { DailyCommentModal } from "../CommentModal";
+import { getAllActiveCommentsOfType, createNewComment } from "../utils";
 
 interface CommentProps {
-    comments: CommentSchema[] | undefined; 
-    setComment: Function; 
-} 
-
-export function CommentCell(props:CommentProps) {
-
-    const [comments, setComments] = useState(props.comments); 
-
-    //TODO - Eventually refactor to handle multiple comments / process for grabbing user 
-    useEffect(() => {
-        // If empty, create stubbed empty one to leave our comments at - eventually refactor to just add this to the end as the current users comments?
-        if (props.comments === undefined) {
-            //Create one empty comment 
-            setComments([{
-                AuthorID:"<TODO Fill this in at some point>", 
-                Type: CommentType.Comment, 
-                Timestamp: moment().tz(TIMEZONE).unix(), 
-                Content:"" 
-            }]); 
-        } 
-    }, []);  
-    
-
-    const onUpdate = (event) => {
-        comments[0].Content = event.target.value; 
-        props.setComment("Comment", comments); 
-    } 
-
-    if (comments !== undefined && comments.length > 0) {
-        return <input defaultValue={comments[0].Content} onChange={onUpdate} />
-    } else {
-        return <input placeholder="Any comments?" onChange={onUpdate} />
+  comments: CommentSchema[] | undefined;
+  setComment: Function; // TODO: fix type
+}
+
+// setComments can be used for comments or reports
+interface ShowCommentModalProps {
+  comments: CommentSchema[];
+  currentComment: CommentSchema;
+  setComments: Function;
+  icon; // TODO: add type
+  color: Color;
+  isEditable: boolean;
+}
+
+function ShowCommentModal({
+  comments,
+  currentComment,
+  setComments,
+  icon,
+  color,
+  isEditable,
+}: ShowCommentModalProps) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [displayedComment, setDisplayedComment] = useState(
+    currentComment
+  );
+  const user = useContext(UserContext);
+
+  const EditableControls = () => {
+    const {
+      isEditing,
+      getSubmitButtonProps,
+      getCancelButtonProps,
+      getEditButtonProps,
+    } = useEditableControls();
+
+    // TODO: change this later to reflect figma
+    return isEditing ? (
+      <ButtonGroup justifyContent="center" size="sm">
+        <Button leftIcon={<CheckIcon />} {...getSubmitButtonProps()} />
+        <Button leftIcon={<CloseIcon />} {...getCancelButtonProps()} />
+      </ButtonGroup>
+    ) : (
+      <Flex justifyContent="right">
+        <Button size="sm" leftIcon={<EditIcon />} {...getEditButtonProps()} />
+      </Flex>
+    );
+  };
+
+  const elevatedUserPrivileges =
```

Comment:
An enum may be useful for these user types to make it more extensible in the future